stream-chat-android-core/src/main/java/io/getstream/chat/android/core/internal/fsm/FiniteStateMachine.kt

```diff
@@ -52,48 +53,30 @@ public class FiniteStateMachine<S : Any, E : Any>(
         }
     }
 
+    /**
+     * The current state as [StateFlow].
+     */
+    public val stateFlow: StateFlow<S> = _state
+
     /**
      * The current state.
      */
     public val state: S
         get() = runBlocking {
-            mutex.withLockIfNotLocked { _state }
+            mutex.withLockIfNotLocked { _state.value }
         }
 
     /**
      * Sends an event to the state machine. The entry point to change state.
      */
     public fun sendEvent(event: E) {
         runBlocking {
-            val shouldNotify = mutex.withLock {
-                val oldState = _state
-                val functions = stateFunctions[oldState::class]
-                val handler = functions?.getHandler(event) ?: defaultEventHandler
-                _state = handler(oldState, event)
-                _state != oldState
-            }
-            if (shouldNotify) {
-                with(_state) {
-                    notifyOnEnter(event)
```

Comment:
The `StateMachine` doesn't need to notify when it moves from one state to a different one, Its mission is only to move from one state to the following one depending on the event it receives. If anyone needs to know that after an event sent to our state machine, it keeps the same state, he can check/verify it when it is being observed